src/Chapter2.hs

```diff
@@ -858,8 +877,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
+rewind :: [a] -> [a]
+rewind [] = []
+rewind (x:xs) = rewind xs ++ [x]
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
am aware of that, just thought it would be easier to understand to write it in this way lol
will attempt a rewrite

btw, am kinda stuck with Chapter3 as it is a lot more free-form
can i submit incomplete pull request?

Reply:
I suppose this is something I am more familiar with but prolly not something you are looking for `rewind xs = foldl (flip (:)) [] xs
`